Riot/Modules/Room/RoomViewController.m

```diff
@@ -3143,6 +3143,45 @@ - (BOOL)dataSource:(MXKDataSource *)dataSource shouldDoAction:(NSString *)action
                             shouldDoAction = NO;
                             break;
                         default:
+                        {
+                            MXEvent *tappedEvent = userInfo[kMXKRoomBubbleCellEventKey];
+                            NSString *format = tappedEvent.content[@"format"];
+                            NSString *formattedBody = tappedEvent.content[@"formatted_body"];
+                            //  if an html formatted body exists
+                            if ([format isEqualToString:kMXRoomMessageFormatHTML] && formattedBody)
+                            {
+                                FormattedBodyParser *parser = [[FormattedBodyParser alloc] initWithFormattedBody:formattedBody];
+                                NSURL *visibleURL = [parser getVisibleURLForURL:url];
+                                
+                                if (visibleURL && ![url isEqual:visibleURL])
```

Comment:
We can make a method like:
`- (BOOL)isVisibleLinkFakedForURL:(NSURL*)url inEvent:(MXEvent*)event;`

An we can also use a property to retain the `FormattedBodyParser`, especially if we are now using a Regex.

Reply:
A boolean returning method won't work for us, because we'll need the visible url to show in the alert.